src/Components/Components/src/ComponentBase.cs

```diff
@@ -41,6 +42,29 @@ public ComponentBase()
         };
     }
 
+    /// <summary>
+    /// Gets the <see cref="ComponentPlatform"/> the component is running on.
+    /// </summary>
+    protected ComponentPlatform Platform => _renderHandle.Platform;
+
+    /// <summary>
+    /// Gets the target <see cref="IComponentRenderMode"/> this component will run in after prerendering.
+    /// </summary>
+    protected IComponentRenderMode? AssignedRenderMode
+    {
+        get
+        {
+            if (!_renderMode.cached)
+            {
+                // We use the platform render mode to avoid having to look up the render mode when the platform
+                // has a single render mode.
```

Comment:
```suggestion
                // We use the platform render mode to avoid having to look up the render mode when the platform
                // has a single render mode.
```

No longer applicable